### PR TITLE
Add inner message definition

### PR DESCRIPTION
**Pull request by buildbreaker**

### CODE REVIEW

File: wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt [hunk 1]
```diff
@@ -189,23 +193,22 @@ fun parseType(field: DescriptorProtos.FieldDescriptorProto): String {
     // Collapsing enums to messages for now
     DescriptorProtos.FieldDescriptorProto.Type.TYPE_ENUM,
     DescriptorProtos.FieldDescriptorProto.Type.TYPE_MESSAGE -> {
-      if (field.typeName.startsWith('.')) {
-        return field.typeName.substring(1)
-      }
-      field.typeName
+      // Get the last section of the type name.
+      // E.g. .bufbuild.testing.NestedMsg.InnerMsg will be InnerMsg
+      field.typeName.split(".").last()
```
Comment (jhump) [inline wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt]: What was here before looks correct. Without the full name, you can't distinguish fields with type `foo.bar.Baz` and `foo.bar.buzz.Baz`, for example.

Comment (jhump) [reply wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt]: Actually, I might see the problem. We are using the parser's "element" types, which are the AST. For that, we actually _need_ the leading dot on all names, or else we might confuse the linker.

So I think you can greatly simplify:
```suggestion
      field.typeName
```

Comment (buildbreaker) [reply wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt]: TY!

**Review by jhump — APPROVED** [review]
LGTM

**Pull request merged**